lib/services/amazonka-dynamodb/src/Amazonka/DynamoDB/Internal.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StrictData #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wall -Werror #-}
+
+-- |
+-- Module      : Amazonka.S3.Internal
+-- Copyright   : (c) 2013-2021 Brendan Hay
+-- License     : Mozilla Public License, v. 2.0.
+-- Maintainer  : Brendan Hay <brendan.g.hay@gmail.com>
+-- Stability   : experimental
+-- Portability : non-portable (GHC extensions)
+module Amazonka.DynamoDB.Internal
+  ( -- * Attribute Values
+    AttributeValue (..),
+
+    -- ** Attribute Value Prisms
+    _L,
+    _NS,
+    _M,
+    _NULL,
+    _N,
+    _BS,
+    _B,
+    _SS,
+    _S,
+    _BOOL,
+  )
+where
+
+import Amazonka.Core
+import Amazonka.Prelude
+import Control.Lens.TH (makePrisms)
+import Data.Aeson
+
+#if MIN_VERSION_aeson(2,0,0)
+import qualified  Data.Aeson.KeyMap as KeyMap
+#else
+import qualified  Data.HashMap.Strict as KeyMap
+#endif
+
+-- | Represents the data for an attribute.
+--
+-- DynamoDB sends and receives JSON objects which contain a single
+-- item whose key is a data type and the value is the data itself. We
+-- provide an actual sum type to interact with these.
+--
+-- For more information, see
+-- <https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes Data Types>
+-- in the /Amazon DynamoDB Developer Guide/.
+data AttributeValue
+  = -- | An attribute of type List. For example:
+    --
+    -- @\"L\": [ {\"S\": \"Cookies\"} , {\"S\": \"Coffee\"}, {\"N\", \"3.14159\"}]@
+    L [AttributeValue]
+  | -- | An attribute of type Number Set. For example:
+    --
+    -- @\"NS\": [\"42.2\", \"-19\", \"7.5\", \"3.14\"]@
+    --
+    -- Numbers are sent across the network to DynamoDB as strings, to maximize
+    -- compatibility across languages and libraries. However, DynamoDB treats
+    -- them as number type attributes for mathematical operations.
+    NS [Text]
+  | -- | An attribute of type Map. For example:
+    --
+    -- @\"M\": {\"Name\": {\"S\": \"Joe\"}, \"Age\": {\"N\": \"35\"}}@
+    M (HashMap Text AttributeValue)
+  | -- | An attribute of type Null. For example:
+    --
+    -- @\"NULL\": true@
+    NULL
+  | -- | An attribute of type Number. For example:
+    --
+    -- @\"N\": \"123.45\"@
+    --
+    -- Numbers are sent across the network to DynamoDB as strings, to maximize
+    -- compatibility across languages and libraries. However, DynamoDB treats
+    -- them as number type attributes for mathematical operations.
+    N Text
```

Comment:
I would unpack this, saves a word and an indirection per text.

Reply:
Thought hard about this, but as soon as you use it anywhere else it'll need to be reboxed, and it's quite likely to get pulled out of the AST and sent elsewhere.